src/d/semantic/caster.d

```diff
@@ -299,7 +299,9 @@ struct Caster(bool isExplicit, alias bailoutOverride = null) {
 					return CastKind.UPad;
 				}
 
-				return CastKind.Invalid;
+				return isFloat(bt)
+					? CastKind.UnsignedToFloat
+					: CastKind.Invalid;
```

Comment:
Not a huge deal, but I don't think a ternary is the best approach here. The code has a "if this do that, if this do that, if this do that, if nothing matches then return invalid" kind of structure.

This means you can add/change/remove the previous branches independently without the overall structure of the code changing, which is what you get with the if and then return invalid. With a ternary, the "default" option is commingled with the last pattern you match. Not a blocker, but this is the kind of details that makes for elegant subsequent diffs when things change.